server/coordinator/queue/delay_queue.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package queue
+
+import (
+	"container/heap"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/CeresDB/ceresmeta/server/coordinator/procedure"
+	"github.com/pkg/errors"
+)
+
+type ProcedureScheduleEntry struct {
+	procedure procedure.Procedure
+	runAfter  time.Time
+}
+
+type ProcedureDelayQueue struct {
+	lock      sync.RWMutex
+	heapQueue *HeapPriorityQueue
+	index     map[uint64]procedure.Procedure
```

Comment:
It looks better if put it as:
```suggestion
	existingProcs     map[uint64]struct{}
```